### PR TITLE
{lang}[foss/2022b] RStudio-Server v2023.03.0+386

### PR DESCRIPTION
**Pull request by Micket**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 1 out of 2 (1 easyconfigs in total)
easybuild-c7.novalocal - Linux CentOS Linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/verdurin/a551292f815203ee7fe638f5be78b438 for a full test report.

**Comment by verdurin** [comment]
@Micket problem with the patch file `RStudio-Server-2022.07.1+554_allow-disabling-quarto.patch`?

**Comment by Micket** [comment]
Sorry, forgot to mark this draft

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Failed during parsing of the easyconfigs, so no ecs were built (1 easyconfigs in total)
vera-c1 - Linux Rocky Linux 8.6, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/Micket/5d024638485671491ddd87220c2a10c8 for a full test report.

File: easybuild/easyconfigs/r/RStudio-Server/RStudio-Server-2023.03.0+386-foss-2022b-Java-11-R-4.2.2.eb [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    ""
+])
+
+configopts = " ".join([
+    "-DRSTUDIO_TOOLS_ROOT='%(builddir)s'",
+    "-DRSTUDIO_TARGET=Server",
+    "-DRSTUDIO_USE_SYSTEM_BOOST=ON",
+    "-DRSTUDIO_USE_SYSTEM_SOCI=ON",
+    "-DRSTUDIO_USE_SYSTEM_YAML_CPP=ON",
+    "-DQUARTO_ENABLED=OFF",  # Not available on all archs, use pandoc fallback
```
Comment (ocaisa) [inline easybuild/easyconfigs/r/RStudio-Server/RStudio-Server-2023.03.0+386-foss-2022b-Java-11-R-4.2.2.eb:70]: In https://github.com/easybuilders/easybuild-easyconfigs/pull/15524/files#diff-71ad7363a3fabd15a4e213d910c9a091417d9e9672ae0430a4156ca95818cce6R36 I think I had an approach that allowed you to have Quarto on the platforms that supported it.

Comment (ocaisa) [reply easybuild/easyconfigs/r/RStudio-Server/RStudio-Server-2023.03.0+386-foss-2022b-Java-11-R-4.2.2.eb:70]: Actually, looking at https://github.com/rstudio/rstudio/blob/8a99de056035023c478eb079ca4a4a50293e9b69/dependencies/common/install-quarto#L66 I think Quarto is now supported for all the platforms we are interested in?

Comment (verdurin) [reply easybuild/easyconfigs/r/RStudio-Server/RStudio-Server-2023.03.0+386-foss-2022b-Java-11-R-4.2.2.eb:70]: I tried commenting that line out, and it failed at the `configure` stage.

**Comment by verdurin** [comment]
Build error:
```
gwtc:

panmirror:
     [echo] yarn location: yarn
     [echo] panmirror location: ./lib/quarto/apps/panmirror
    [mkdir] Created dir: /dev/shm/RStudioServer/2023.03.0+386/foss-2022b-Java-11-R-4.2.2/rstudio-2023.03.0-386/src/gwt/www/js/panmirror

BUILD FAILED
/dev/shm/RStudioServer/2023.03.0+386/foss-2022b-Java-11-R-4.2.2/rstudio-2023.03.0-386/src/gwt/build.xml:173: The following error occurred while executing this line:
/dev/shm/RStudioServer/2023.03.0+386/foss-2022b-Java-11-R-4.2.2/rstudio-2023.03.0-386/src/gwt/build.xml:132: The directory /dev/shm/RStudioServer/2023.03.0+386/foss-2022b-Java-11-R-4.2.2/rstudio-2023.03.0-386/src/gwt/lib/quarto/apps/panmirror does not exist
```

File: easybuild/easyconfigs/r/RStudio-Server/RStudio-Server-2023.03.0+386-foss-2022b-Java-11-R-4.2.2.eb [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+If you need a database config one can be created with:
+  MYTMP=`mktemp -d` && echo -e "provider=sqlite\\ndirectory=${MYTMP}/sqlite" > "${MYTMP}/db.conf"
+and then used with:
+  rserver ... --database-config-file="${MYTMP}/db.conf"
+"""
+
+toolchain = {'name': 'foss', 'version': '2022b'}
+
+source_urls = ['https://github.com/rstudio/rstudio/archive']
+sources = ['v%(version)s.tar.gz']
```
Comment (ocaisa) [inline easybuild/easyconfigs/r/RStudio-Server/RStudio-Server-2023.03.0+386-foss-2022b-Java-11-R-4.2.2.eb:24]: Are we missing this patch? https://github.com/ComputeCanada/easybuild-easyconfigs/blob/computecanada-main/easybuild/easyconfigs/r/RStudio-Server/rstudio-1.2.1335.patch

**Comment by Micket** [comment]
I won't be touching this anymore. Feel free to take over this and do whatever you want.

**Pull request closed without merging**